modules/katello_content_view.py

```diff
@@ -77,79 +85,32 @@
+        create_server,
+        ping_server,
+        find_organization,
+        find_content_view,
+        find_repositories,
+        naildown_entity_state,
+        sanitize_entity_dict,
+    )
+
     has_import_error = False
 except ImportError as e:
     has_import_error = True
     import_error_msg = str(e)
 
 from ansible.module_utils.basic import AnsibleModule
 
-
-class NailGun(object):
-    def __init__(self, server, entities, module):
-        self._server = server
-        self._entities = entities
-        self._module = module
-
-    def find_organization(self, name):
-        org = self._entities.Organization(self._server, name=name)
-        response = org.search(set(), {'search': 'name="{}"'.format(name)})
-
-        if len(response) == 1:
-            return response[0]
-        else:
-            self._module.fail_json(msg="No organization found for %s" % name)
-
-    def find_product(self, name, organization):
-        product = self._entities.Product(self._server, name=name, organization=organization)
-        response = product.search()
-
-        if len(response) == 1:
-            return response[0]
-        else:
-            self._module.fail_json(msg="No Product found for %s" % name)
-
-    def find_repository(self, name, product, organization):
-        product = self.find_product(product, organization)
-
-        repository = self._entities.Repository(self._server, name=name, product=product)
-        repository._fields['organization'] = entity_fields.OneToOneField(entities.Organization)
-        repository.organization = product.organization
-        response = repository.search()
-
-        if len(response) == 1:
-            return response[0]
-        else:
-            self._module.fail_json(msg="No Repository found for %s" % name)
-
-    def find_repositories(self, repositories, organization):
-        return map(lambda repository: self.find_repository(repository['name'], repository['product'], organization), repositories)
-
-    def content_view(self, name, organization, repositories=[]):
-        updated = False
-        organization = self.find_organization(organization)
-
-        content_view = self._entities.ContentView(self._server, name=name, organization=organization)
-        response = content_view.search()
-
-        if len(response) == 1:
-            content_view = response[0]
-        elif len(response) == 0:
-            if not self._module.check_mode:
-                content_view = content_view.create()
-            updated = True
-
-        repositories = self.find_repositories(repositories, organization)
-
-        if set(map(lambda r: r.id, repositories)) != set(map(lambda r: r.id, content_view.repository)):
-            if not self._module.check_mode:
-                content_view.repository = repositories
-                content_view.update(['repository'])
-            updated = True
-
-        return updated
+content_view_map = {
```

Comment:
I would prefer to have `name_map` in this identifier, because it maps names from the ansible module to the nailgun-interface.